### ISSUE TITLE
Could not find key /_1 in the model file even though the key is there

### ISSUE THREAD

**Issue by danielhers** [clab/dynet#1580]
To reproduce:
```
import dynet as dy

m = dy.ParameterCollection()
a = m.add_lookup_parameters((100, 100))
b = m.add_parameters(100)
t = dy.SimpleSGDTrainer(m)
e = dy.sum_elems(a)
e.forward()
(e/0).backward()
try:
    t.update()
except RuntimeError as e:
    print(e)
dy.save("/tmp/tmp.model", (a, b))

m2 = dy.ParameterCollection()
x = m2.add_lookup_parameters((100, 100))
y = m2.add_parameters(100)
dy.load("/tmp/tmp.model", m2)
```
I wrapped the `update()` in a try/except because I don't want the whole program to die just because one gradient is bad. It's definitely a problem that all updates fail because of bad gradients, but this error message did not help me realize this is the case.
Did I do something wrong, or is this a bug? It happened in my project using DyNet and it took a long time to figure out why (danielhers/tupa#79).

**Comment by neubig** [comment]
This looks like a bug.
